SDL/TrackCandidate.cuh

```diff
@@ -42,25 +40,402 @@ namespace SDL
 
     void createTrackCandidatesInExplicitMemory(struct trackCandidates& trackCandidatesInGPU, unsigned int maxTrackCandidates,cudaStream_t stream);
     
-    ALPAKA_FN_ACC void addTrackCandidateToMemory(struct trackCandidates& trackCandidatesInGPU, short trackCandidateType, unsigned int innerTrackletIndex, unsigned int outerTrackletIndex, unsigned int trackCandidateIndex, unsigned int directObjectIndex);
+    ALPAKA_FN_ACC ALPAKA_FN_INLINE void addTrackCandidateToMemory(struct trackCandidates& trackCandidatesInGPU, short trackCandidateType, unsigned int innerTrackletIndex, unsigned int outerTrackletIndex, unsigned int trackCandidateIndex, unsigned int directObjectIndex)
+    {
+        trackCandidatesInGPU.trackCandidateType[trackCandidateIndex] = trackCandidateType;
+        trackCandidatesInGPU.directObjectIndices[trackCandidateIndex] = directObjectIndex;
+        trackCandidatesInGPU.objectIndices[2 * trackCandidateIndex] = innerTrackletIndex;
+        trackCandidatesInGPU.objectIndices[2 * trackCandidateIndex + 1] = outerTrackletIndex;
+    };
 
-    ALPAKA_FN_ACC void addTrackCandidateToMemory(struct trackCandidates& trackCandidatesInGPU, short trackCandidateType, unsigned int innerTrackletIndex, unsigned int outerTrackletIndex, uint8_t* logicalLayerIndices, uint16_t* lowerModuleIndices, unsigned int* hitIndices, float centerX, float centerY, float radius, unsigned int trackCandidateIndex, unsigned int directObjectIndex);
+    ALPAKA_FN_ACC ALPAKA_FN_INLINE void addTrackCandidateToMemory(struct trackCandidates& trackCandidatesInGPU, short trackCandidateType, unsigned int innerTrackletIndex, unsigned int outerTrackletIndex, uint8_t* logicalLayerIndices, uint16_t* lowerModuleIndices, unsigned int* hitIndices, float centerX, float centerY, float radius, unsigned int trackCandidateIndex, unsigned int directObjectIndex)
```

Comment:
I think that we should delay simplifying this code until the alpaka branch is pushed in, because I didn't write this function so I'm not sure what the best way to simplify it would be (to make it more clear and reduce function inputs). For example, the overloaded pLS function code is also present (copy-pasted) in the more general function, so I'm not sure if it's even necessary to keep it around. Maybe we can just add this to the growing list of things that should be improved later on?

I will rename it though like you mentioned to avoid having functions with duplicate names.

Reply:
I agree with the plan, thanks!